contrib/olivere/elastic/elastictrace.go

```diff
@@ -134,8 +134,8 @@ func peek(rc io.ReadCloser, encoding string, max, n int) (string, io.ReadCloser,
 	}
 	if encoding == "gzip" {
 		// unpack the snippet
-		gzr, err := gzip.NewReader(bytes.NewReader(snip))
-		if err != nil {
+		gzr, err2 := gzip.NewReader(bytes.NewReader(snip))
+		if err2 != nil {
```

Comment:
@mrkm4ntr maybe I haven't had enough coffee yet, but I don't understand why the previous assignment would have been "ineffectual"? See https://play.golang.org/p/hUfSbzBrIoG

Reply:
Actual ineffectual assignment is here. The scope of `err` is in this block. So I changed the name of `err` in this scope to `err2`. 
https://github.com/DataDog/dd-trace-go/blob/f86a82b0ae679be3bbd2fe3652ae17f06aabd960/contrib/olivere/elastic/elastictrace.go#L143

Reply:
Got it. Subtle. LGTM!

Reply:
I don't see how this changes the logic. Whether the variable is named `err` or `err2`, it is block-scoped and never used outside the block. Likewise, we're not using the outer `err` inside the block, so shadowing is not an issue.

If there is a bug here, please add a test that exposes the bug which is fixed by this change.

Reply:
Again, the problematic part is here.
https://github.com/DataDog/dd-trace-go/blob/f86a82b0ae679be3bbd2fe3652ae17f06aabd960/contrib/olivere/elastic/elastictrace.go#L143-L145
if `ioutil.ReadAll(gzr)` returns error, that error is assigned to scoped `err` not `err` that function returns. So this function cannot return error of `ioutil.ReadAll(gzr)`.

The test is still required...? I think it is preferred to introduce linters to detect these kind of bugs automatically.
https://github.com/gordonklaus/ineffassign

Reply:
@mrkm4ntr Thank you for the clarification. I did not look closely enough my first time through.

A test case is always welcome, but I think we can merge this as-is.